discussions.go

```diff
@@ -0,0 +1,477 @@
+	}
+
+	return n, resp, err
+}
+
+// DeleteSnippetDiscussion deletes an existing discussion of a snippet.
+//
+// https://docs.gitlab.com/ce/api/discussions.html#delete-a-snippet-discussion
+func (s *DiscussionsService) DeleteSnippetDiscussion(pid interface{}, snippet, discussion int, options ...OptionFunc) (*Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, err
+	}
+	u := fmt.Sprintf("projects/%s/snippets/%d/discussions/%d", url.QueryEscape(project), snippet, discussion)
+
+	req, err := s.client.NewRequest("DELETE", u, nil, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(req, nil)
+}
+
+// ListMergeRequestDiscussionsOptions represents the available ListMergeRequestDiscussions()
+// options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#list-all-merge-request-discussions
+type ListMergeRequestDiscussionsOptions ListOptions
+
+// ListMergeRequestDiscussions gets a list of all discussions for a single merge request.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#list-all-merge-request-discussions
+func (s *DiscussionsService) ListMergeRequestDiscussions(pid interface{}, mergeRequest int, opt *ListMergeRequestDiscussionsOptions, options ...OptionFunc) ([]*Discussion, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/merge_requests/%d/discussions", url.QueryEscape(project), mergeRequest)
+
+	req, err := s.client.NewRequest("GET", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var n []*Discussion
+	resp, err := s.client.Do(req, &n)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return n, resp, err
+}
+
+// GetMergeRequestDiscussion returns a single discussion for a given merge request.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#get-single-merge-request-discussion
+func (s *DiscussionsService) GetMergeRequestDiscussion(pid interface{}, mergeRequest, discussion int, options ...OptionFunc) (*Discussion, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/merge_requests/%d/discussions/%d", url.QueryEscape(project), mergeRequest, discussion)
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	n := new(Discussion)
+	resp, err := s.client.Do(req, n)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return n, resp, err
+}
+
+// CreateMergeRequestDiscussionOptions represents the available
+// CreateMergeRequestDiscussion() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#create-new-merge-request-discussion
+type CreateMergeRequestDiscussionOptions struct {
+	Body      *string      `url:"body,omitempty" json:"body,omitempty"`
+	CreatedAt time.Time    `url:"created_at,omitempty"`
+	Position  NotePosition `json:"position,omitempty"`
```

Comment:
All fields should have the `url` and `json` tags, and both `time.Time` and `NotePosition` should be a pointer.